authentication/account/service/user.go

```diff
@@ -0,0 +1,127 @@
+func NewUserService(ctx servicecontext.ServiceContext) service.UserService {
+	return &userServiceImpl{
+		BaseService: base.NewBaseService(ctx),
+	}
+}
+
+// userServiceImpl implements the UserService to manage users
+type userServiceImpl struct {
+	base.BaseService
+}
+
+// UserInfo gets user information given a context containing access_token
+func (s *userServiceImpl) UserInfo(ctx context.Context, identityID uuid.UUID) (*repository.User, *repository.Identity, error) {
+	var identity *repository.Identity
+	err := s.ExecuteInTransaction(func() error {
+		var err error
+		identity, err = s.Repositories().Identities().LoadWithUser(ctx, identityID)
+		if err != nil || identity == nil {
+			return errors.NewUnauthorizedError(fmt.Sprintf("auth token contains id %s of unknown Identity\n", identityID))
+		}
+		return nil
+	})
+
+	if err != nil {
+		return nil, nil, err
+	}
+	log.Debug(ctx, map[string]interface{}{
+		"identity_id": identity.ID,
+		"user_id":     identity.User.ID,
+	}, "loaded identity and user")
+	return &identity.User, identity, nil
+}
+
+func (s *userServiceImpl) DeprovisionUser(ctx context.Context, username string) (*repository.Identity, error) {
+
+	var identity *repository.Identity
+	err := s.ExecuteInTransaction(func() error {
+
+		identities, err := s.Repositories().Identities().Query(
+			repository.IdentityWithUser(),
+			repository.IdentityFilterByUsername(username),
+			repository.IdentityFilterByProviderType(repository.DefaultIDP))
+		if err != nil {
+			return err
+		}
+		if len(identities) == 0 {
+			return errors.NewNotFoundErrorWithKey("user identity", "username", username)
+		}
+
+		identity = &identities[0]
+		identity.User.Deprovisioned = true
+
+		return s.Repositories().Users().Save(ctx, &identity.User)
+	})
+
+	return identity, err
+}
+
+// ContextIdentityIfExists returns the identity's ID found in given context if the identity exists in the Auth DB
+// If it doesn't exist then an Unauthorized error is returned
+func (s *userServiceImpl) ContextIdentityIfExists(ctx context.Context) (uuid.UUID, error) {
+	identity, err := manager.ContextIdentity(ctx)
+	if err != nil {
+		return uuid.Nil, err
+	}
+	// Check if the identity exists
+	err = s.ExecuteInTransaction(func() error {
+		err := s.Repositories().Identities().CheckExists(ctx, identity.String())
+		if err != nil {
+			return errors.NewUnauthorizedError(err.Error())
+		}
+		return nil
+	})
+	if err != nil {
+		return uuid.Nil, err
+	}
+	return *identity, nil
+}
+
+// LoadContextIdentityAndUser returns the identity found in given context if the identity exists in the Auth DB
+// If no token present in the context then an Unauthorized error is returned
+// If the identity represented by the token doesn't exist in the DB or not associated with any User then an Unauthorized error is returned
+func (s *userServiceImpl) LoadContextIdentityAndUser(ctx context.Context) (*repository.Identity, error) {
+	var identity *repository.Identity
```

Comment:
no need to declare the variable, just use `identity, err := ...` on line 108

Reply:
This is a harmless change, fixed.